pkg/flow/adapter/dataweavetransformation/adapter.go

```diff
@@ -96,49 +115,95 @@ func (a *dataweaveTransformAdapter) Start(ctx context.Context) error {
 
 func (a *dataweaveTransformAdapter) dispatch(ctx context.Context, event cloudevents.Event) (*cloudevents.Event, cloudevents.Result) {
 	var err error
-	var tmpfile *os.File
+	var tmpFile *os.File
 	path := "/tmp/dw"
 	dwFolder := path + "/custom"
 
-	switch a.incomingContentType {
-	case "application/json":
-		if !isValidJSON(event.Data()) {
-			return a.replier.Error(&event, targetce.ErrorCodeRequestParsing,
-				errors.New("invalid Json"), nil)
+	err = validateContentType(event.DataContentType(), event.Data())
+	if err != nil {
+		return a.replier.Error(&event, targetce.ErrorCodeRequestValidation,
+			errors.New(err.Error()), nil)
+	}
+
+	req := &DataWeaveTransformationStructuredRequest{}
+	if err := event.DataAs(req); err != nil {
+		return a.replier.Error(&event, targetce.ErrorCodeRequestParsing, err, nil)
+	}
+
+	if !a.spellOverride && req.Spell != "" {
+		return a.replier.Error(&event, targetce.ErrorCodeRequestValidation,
+			errors.New("it is not allowed to override Spell per CloudEvent"), nil)
+	}
+
+	if a.defaultSpell == nil && req.Spell == "" {
+		return a.replier.Error(&event, targetce.ErrorCodeRequestValidation,
+			errors.New("no default Spell or in request found"), nil)
+	}
+
+	if req.InputContentType != "" || req.OutputContentType != "" || req.Spell != "" {
+		if req.InputData == "" {
+			return a.replier.Error(&event, targetce.ErrorCodeRequestValidation,
+				errors.New("inputData not found"), nil)
+		}
+	}
+
+	// Check for spellOverride to be enabled and all the parameters to be present.
+	if a.spellOverride && req.InputData != "" && req.InputContentType != "" && req.OutputContentType != "" && req.Spell != "" {
+		inputData = []byte(req.InputData)
```

Comment:
I see a huge code smell here.
We rely on a global state but the code is concurrent, so multiple goroutines will be writing this value at the same time.

Reply:
It should be fixed now